src/state/slices/opportunitiesSlice/selectors.test.ts

```diff
@@ -0,0 +1,272 @@
+          'eip155:1:0xLpContractOne': '424242',
+        },
+      },
+      ids: ['eip155:1:0xgomes', 'eip155:1:fauxmes'],
+    }
+    const lp = {
+      ...initialState.lp,
+      byAccountId: {
+        'eip155:1:0xgomes': ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+        'eip155:1:fauxmes': ['eip155:1:0xLpContractOne'],
+      },
+      ids: ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+    }
+    const staking = {
+      ...initialState.staking,
+      byAccountId: {
+        'eip155:1:0xgomes': ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+        'eip155:1:fauxmes': ['eip155:1:0xLpContractOne'],
+      },
+      ids: ['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'],
+    }
+    const userStaking = {
+      ...initialState.staking,
+      ids: [
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractTwo'),
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractOne'),
+        serializeUserStakingId('eip155:1:0xfauxmes', 'eip155:1:0xLpContractOne'),
+      ] as UserStakingId[],
+      byId: {
+        [serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractTwo')]: {
+          stakedAmountCryptoPrecision: '1337',
+          rewardsAmountCryptoPrecision: '420',
+        },
+        [serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractOne')]: {
+          stakedAmountCryptoPrecision: '4',
+          rewardsAmountCryptoPrecision: '3',
+        },
+        [serializeUserStakingId('eip155:1:0xfauxmes', 'eip155:1:0xLpContractOne')]: {
+          stakedAmountCryptoPrecision: '9000',
+          rewardsAmountCryptoPrecision: '1',
+        },
+      },
+    }
+
+    const mockState = {
+      ...baseState,
+      portfolio: {
+        ...baseState.portfolio,
+        accountBalances,
+      },
+      opportunities: {
+        ...initialState,
+        lp,
+        staking,
+        userStaking,
+      },
+    }
+    describe('selectLpOpportunityIdsByAccountId', () => {
+      it('can get LP opportunities Ids for a given AccountId', () => {
+        const result = selectLpOpportunityIdsByAccountId(mockState, {
+          accountId: 'eip155:1:0xgomes',
+        })
+
+        expect(result).toEqual(['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'])
+      })
+    })
+    describe('selectStakingOpportunityIdsByAccountId', () => {
+      it('can get staking opportunities Ids for a given AccountId', () => {
+        const result = selectStakingOpportunityIdsByAccountId(mockState, {
+          accountId: 'eip155:1:0xgomes',
+        })
+
+        expect(result).toEqual(['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'])
+      })
+    })
+    describe('selectHighestBalanceLpUserStakingIdByStakingId', () => {
+      it('can get the highest balance AccuontId for a given StakingId', () => {
+        const result = selectHighestBalanceAccountIdIdByStakingId(mockState, {
+          stakingId: 'eip155:1:0xLpContractOne',
```

Comment:
should these not be a farming/staking contract over an lp one?

Reply:
Definitely, after discussing this one out, way more sane to have the actual IDs of either farming | staking | serialized AssetIdish of whatever we might support in the future - 45a0d8ce8edc118a158a6ab80121bd3ffe5e4126